rust/main/submitter/src/chain_tx_adapter/chains/ethereum/adapter/tx_status_checker.rs

```diff
@@ -0,0 +1,52 @@
+// the evm provider-building logic returns a box. `EvmProviderForSubmitter` is only implemented for the underlying type rather than the boxed type.
+// implementing the trait for the boxed type would require a lot of boilerplate code.
+#![allow(clippy::borrowed_box)]
+
+use ethers::types::U64;
+use hyperlane_ethereum::{EthereumReorgPeriod, EvmProviderForSubmitter};
+use tracing::warn;
+
+use crate::{SubmitterError, TransactionStatus};
+
+async fn block_number_result_to_tx_hash(
```

Comment:
It looks like this function maps block number of a transaction to transaction status. The name of the function is a little bit confusing.

Reply:
meant that to be `block_number_result_to_tx_status`, good catch